debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -205,9 +208,11 @@ public void changeRecord(DataCollectionSchema schema,
                                              OffsetContext offset,
                                              ConnectHeaders headers)
                             throws InterruptedException {
-                        transactionMonitor.dataEvent(dataCollectionId, offset, key, value);
-                        eventListener.onEvent(dataCollectionId, offset, key, value);
-                        streamingReceiver.changeRecord(schema, operation, key, value, offset, headers);
+                        if (!skippedOperations.contains(operation)) {
```

Comment:
@blcksrx Am I right you said we can solve filtering more performant with db tools for some databases (like SQL Server for example)?
@jpechane you suggested a design that for such cases we can use the more performant solution instead. what was your idea about we make this possible? (like how to deactivate this global behavior here in core in favor of the more performant implementation in a connector implementation?

Reply:
Yes, you are right @jpechane suggest first implement it on the core, so this feature will be available for all the connector. after that we can design it for each connector with a better performance.

Reply:
I see!
@jpechane do we need to do anything in this PR to address that flexibility now or do we do this as bigger refactoring with the first connector specific implementation?

Reply:
@rk3rn3r Nothing specific needed right now. MySQL is already rewritten to use the new framework and IIRC the skipped ops are already implemente dthere so it might be a good starting point. @blcksrx Could you please raise a follow-up Jira for that?

@blcksrx Also I've one more tip for micro-optimization.  There should be a boolean flag attribute `neverSkip` that is set at the start depending on whether the map is empty or not.

The condition would then look like
```
if (neverSkip ||  !skippedOperations.contains(operation))
```
This would be more performant in case skipped ops are not set.

Reply:
@jpechane thanks for answering. I will do the optimization tonight and about another issue, currently, the `MongoDB connectors try to filter that on the `OpLog` side. I can create tickets for other connectors